### ISSUE TITLE
Tab Inconsistencies

### ISSUE THREAD

**Issue by TodesBrot** [SixLabors/Fonts#31]
When the TabWidth is set to 0, any line containing a tab will be cut off following the tab.

When the TabWidth is greater than 1, the actual drawn TabWidth will be 1 too small.

Measuring strings starting or ending with a tab ignores those tabs.

Measuring the string "x\tx" in a monospaced font returns a width of 4 characters when the TabWidth is 4, so the tab seems to be 2 spaces too short. Measuring "x\t\tx" returns a width of 8 characters, so it seems like one of the tabs has the correct width.

**Comment by tocsoft** [comment]
> When the TabWidth is set to 0, any line containing a tab will be cut off following the tab.

This is a bug and I've split this out into issue #32

> When the TabWidth is greater than 1, the actual drawn TabWidth will be 1 too small.

The way the library is currently setup is that tabs work as a series of Tab stops at tab width boundaries, with a minimum width of one space.

I'll layout a couple of examples to illustrate: where 'X' = 5px & space = 10px and tabwidth = 4 (measured in space widths) therefore tabs will stop every 40px.

so take the text `'x' 'x' 'x' '\t' 'x'` for example based on the measurements above it would end up as 45px wide, this would breaking down to this:
- 5px for the first `x` (moving us to 5px along)
- 5px for the second `x` (moving us to 10px along)
- 5px for the third `x` (moving us to 15px along)
- 25px for the `\t` (moving us along to 40px) as tabs move >= space width but stopping at a 'tab stop'
- 5px for last `x` (moving us to 45px)

now on the other hand `'x' 'x' 'x' '\t' '\t' 'x'` would end up as 85px; 
- 5px for the first `x` (moving us to 5px along)
- 5px for the second `x` (moving us to 10px along)
- 5px for the third `x` (moving us to 15px along)
- 25px for the `\t` (moving us along to 40px) as tabs move >= space width but stopping at a 'tab stop'
- 40px for the `\t` (moving us along to 80px) moveing us one space forward then continuing forward until it hits a tab stop
- 5px for last `x` (moving us to 85px)

Hope the clarifies how calculated tab widths end up being calculated.

Basically its not directly equatable of 1 tab character will always be the same exact width.

In conclusion tabs are currently working as designed (except for bugs as noted), and this tab rendering mechanism is consistent with other text rendering software, Word does it, even Notepad does. 

> Measuring strings starting or ending with a tab ignores those tabs.

This is a bug and I've split this out into issue #33

**Comment by tocsoft** [comment]
So i'm closing this and will track the other 2 issues separately
